core/app/models/spree/promotion.rb

```diff
@@ -120,7 +120,12 @@ def deactivate(payload)
     def eligible?(promotable)
```

Comment:
```suggestion
def eligible?(promotable)
    return false if expired? || usage_limit_exceeded?(promotable) || blacklisted?(promotable)
    return !!eligible_rules(promotable, {}) && !!eligible_rules(promotable.order, {}) if promotable == Spree::LineItem
    
    !!eligible_rules(promotable, {})
end
```

Reply:
`promotable` is not a class, so comparing it to `Spree::LineItem` using `==` won't work.